pkg/cloudprovider/aws/instance.go

```diff
@@ -107,22 +162,54 @@ func (p *InstanceProvider) Create(ctx context.Context,
 	return createFleetOutput.Instances[0].InstanceIds[0], nil
 }
 
-func (p *InstanceProvider) Terminate(ctx context.Context, node *v1.Node) error {
-	id, err := getInstanceID(node)
+func (p *InstanceProvider) getInstance(ctx context.Context, id *string, instance *ec2.Instance) error {
+	describeInstancesOutput, err := p.ec2api.DescribeInstancesWithContext(ctx, &ec2.DescribeInstancesInput{InstanceIds: []*string{id}})
+	if aerr, ok := err.(awserr.Error); ok && aerr.Code() == "InvalidInstanceID.NotFound" {
```

Comment:
I think this is the same constant as in Line 93. Can you change this accordingly?

Reply:
Nice catch!